test/release/examples/primers/loops.chpl

```diff
@@ -0,0 +1,950 @@
+  coforall example above.
+
+  In practice, however, most parallel iterators (including those
+  defined on ranges, domains, and arrays) will take stock of the
+  number of tasks running on the current locale and then throttle the
+  number of tasks they create to avoid overwhelming the node.  As a
+  result, a nested forall-loop over a pair of ranges, like:
+  
+*/
+
+  forall i in 1..500 do
+    forall j in 1..500 do
+      /* do some computation here */;
+
+/*      
+
+  will typically create only ``numCores`` tasks in total.
+  Specifically, the outer loop will create ``numCores`` tasks, then
+  each of the inner loops will see that all the cores are busy and
+  avoid creating additional tasks to avoid creating needless
+  parallelism.
+  
+  In such cases, if the inner loop body was not computationally
+  intensive, it could make sense to rewrite the inner loop as a
+  ``foreach`` in order to avoid the overhead of having the iterator
+  determine whether or not to create tasks at all:
+
+*/
+
+  forall i in 1..500 do
+    foreach j in 1..500 do
+      /* do some computation here */;
+
+/*
+
+  That said, such overheads are relatively modest, so loop bodies that
+  are computationally intense, the benefit for changing the inner loop
+  from ``forall`` to ``foreach`` may be negligible.
+
+  In summary, there is nothing magical about nested loops.  When
+  reasoning about what a given loop nest does, simply consider the
+  loops one at a time.  For example, what does the outer loop do?
+  ("It's a ``forall``, so it will invoke the parallel iterator
+  specified by its iterand.")  OK, what about the next loop?  ("It's a
+  `coforall`, so it will literally create a task per iteration
+  regardless of how many are already running).  What about the next
+  loop?  ("It's a ``foreach``, so it will try to use SIMD/SIMT
+  features in the task's current target processor to implement its
+  iterations").  Chapel's implementation of parallel loops is very
+  imperative, where the most complex case is being familiar with the
+  parallelism implemented by any iterand expressions of a ``forall``
+  loop.
+
+
+  When to Use Which Loop Form?
+  ----------------------------
+
+  Given these various loop forms, which ones should you use when?
+
+  Starting with the obvious, if you have a loop that wants or needs to
+  be serial, such as a loop spanning the time steps of a simulation,
+  you should use one of the serial loop forms.  When writing a serial
+  loop, if you are iterating over a type that supports iteration, like
+  an array, domain, range, or list, the for-loop can often be the
+  clearest and most elegant loop form.  Or, if a serial iterator
+  exists that does what you want, the for-loop is also an obvious
+  choice.  But if you want to do something more general, or unrelated
+  to an iterator, the while-loop can serve as a more general fallback.
+  Alternatively, you can abstract your loop structure into a new
```

Comment:
This primer has a lot of background in it, but probably not enough for someone to understand what this sentence means.

Reply:
I tried to reword for clarity and also re-linked the iterators primer as a reference:

> But if you want to do something more general that is not             
  currently supported by an iterator, the while-loop can serve as a             
  more general fallback.  Or you could write an iterator of your own            
  that wraps your serial loop structure, and then use a for-loop to             
  invoke it.  Refer to the :ref:`Iterators Primer <primers-iterators>`          
  for more information about doing this.